### PR TITLE
[WIP] plugins/main/ptp: Add IPv6 support

### PR DESCRIPTION
**Pull request by fnordahl**
Fixes part of #31

### CODE REVIEW

**Comment by dcbw** [comment]
It seems like Go should have a better mechanism to create a bitmask, but lgtm.

File: plugins/main/ptp/ptp.go [hunk 1]
```diff
@@ -120,26 +127,33 @@ func setupContainerVeth(netns, ifName string, mtu int, pr *types.Result) (string
...
+func setupHostVeth(vethName string, ipamResult *types.Result) error {
 	// hostVeth moved namespaces and may have a new ifindex
 	veth, err := netlink.LinkByName(vethName)
 	if err != nil {
 		return fmt.Errorf("failed to lookup %q: %v", vethName, err)
 	}
 
-	// TODO(eyakubovich): IPv6
+	ipConfig := ipamResult.IP6
+	cidrOnes := 128
```
Comment (steveeJ) [inline plugins/main/ptp/ptp.go]: please declare these magic numbers as const, like so:
```
const (
    CIDROnesIPv6 = 128
    CIDROnesIPv4 = 32
)
```
Of course make sure I understood the names correctly ;-)

File: plugins/main/ptp/ptp.go [hunk 2]
```diff
@@ -120,26 +127,33 @@ func setupContainerVeth(netns, ifName string, mtu int, pr *types.Result) (string
...
+	cidrBits := 128
+	if ipConfig == nil {
```
Comment (steveeJ) [inline plugins/main/ptp/ptp.go]: This behavior will prefer IPv6 over IPv4. Ideally we'd support both simultaneously, but we have discussed in other issues that it probably requires a spec revision. But can be done in this PR already is document the new behavior and then follow-up with a spec change.

Comment (dcbw) [inline plugins/main/ptp/ptp.go]: Maybe even a helper function like:

```golang
func getIpConfig(result types.Result) (*types.IPConfig, uint, uint) {
    <stuff>
}
```

which might be a bit clearer in the code.

**Comment by rosenhouse** [comment]
Plugins have moved to [their own repository](https://github.com/containernetworking/plugins).  If you'd like to continue development, open a new PR over there.

**Pull request closed without merging**